main.tf

```diff
@@ -217,6 +217,10 @@ resource "aws_rds_cluster" "primary" {
   enabled_cloudwatch_logs_exports = var.enabled_cloudwatch_logs_exports
   deletion_protection             = var.deletion_protection
   replication_source_identifier   = var.replication_source_identifier
+
+  lifecycle {
+    ignore_changes = [engine_version]
+  }
```

Comment:
If you want to use auto minor version upgrade, couldn't you set the engine version to a major version with auto minor upgrades enabled so when a new minor version comes out, it won't try to downgrade it ?

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster#engine_version-1

> The value can contain a partial version where supported by the API.

```diff
-  engine_version = "5.7.mysql_aurora.2.07.6"
+  engine_version = "5.7"
```

See https://github.com/hashicorp/terraform-provider-aws/issues/25348#issuecomment-1596923574